gef.py

```diff
@@ -10581,38 +10583,54 @@ def parse_procfs_maps() -> Generator[Section, None, None]:
                               path=pathname)
         return
 
-    @staticmethod
-    def parse_gdb_info_proc_maps() -> Generator[Section, None, None]:
+    def __parse_gdb_info_proc_maps(self) -> Generator[Section, None, None]:
         """Get the memory mapping from GDB's command `maintenance info sections` (limited info)."""
-
         if GDB_VERSION < (11, 0):
             raise AttributeError("Disregarding old format")
 
-        lines = (gdb.execute("info proc mappings", to_string=True) or "").splitlines()
+        output = (gdb.execute("info proc mappings", to_string=True) or "")
+        if not output:
+            raise AttributeError
+
+        start_idx = output.find("Start Addr")
+        if start_idx == -1:
+            raise AttributeError
+
+        output = output[start_idx:]
+        lines = output.splitlines()
+        if len(lines) < 2:
+            raise AttributeError
 
-        # The function assumes the following output format (as of GDB 11+) for `info proc mappings`
+        # The function assumes the following output format (as of GDB 11+) for `info proc mappings`:
+        # - live process (incl. remote)
         # ```
-        # process 61789
-        # Mapped address spaces:
-        #
         #           Start Addr           End Addr       Size     Offset  Perms  objfile
         #       0x555555554000     0x555555558000     0x4000        0x0  r--p   /usr/bin/ls
         #       0x555555558000     0x55555556c000    0x14000     0x4000  r-xp   /usr/bin/ls
         # [...]
         # ```
+        # or
+        # - coredump & rr
+        # ```
+        #          Start Addr           End Addr       Size     Offset objfile
+        #      0x555555554000     0x555555558000     0x4000        0x0 /usr/bin/ls
+        #      0x555555558000     0x55555556c000    0x14000     0x4000 /usr/bin/ls
+        # ```
+        # In the latter case, mock the Permission to `rwx` so `dereference` would still work
 
-        if len(lines) < 5:
-            raise AttributeError
-
-        # Format seems valid, iterate to generate sections
-        for line in lines[4:]:
+        mock_permission = all(map(lambda x: x.strip() != "Perms", lines[0].split()))
```

Comment:
what is this doing? Can you explain in a comment?

Reply:
the comment is right above :D

Reply:
OK. So `mock_permission` is a bool that is true if all lines (Including the header I guess is OK) doesn't have the work `Perms`? But I don't see that in the example at all.

Reply:
The (large) comment above explains the output formats we should parse so both live process and cold debug (coredump, rr) also work.

> So mock_permission is a bool that is true if all lines (Including the header I guess is OK) doesn't have the work Perms

If there's no `Perms` columns from the output, then we mock them. 
If my comment is not clear enough happy to re-phrase.

Reply:
Adjusted